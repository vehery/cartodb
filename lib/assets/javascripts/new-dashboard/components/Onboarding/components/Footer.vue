<template>
  <div class="footer">
    <div class="container u-flex u-flex__justify--end">
      <button class="button button--ghost js-prev" v-if="showPrevButton" @click="prev">
        {{ $t('Wizards.footer.back') }}
      </button>
      <button class="button is-primary js-next" v-if="showNextButton" @click="next">
        {{ $t('Wizards.footer.next', { nextStep: this.stepNames[this.currentStep] }) }}
      </button>
      <button class="button is-primary js-goToDashboard" v-if="!showNextButton" @click="goToDashboard">
        {{ $t('Wizards.footer.goToDashboard') }}
      </button>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Footer',
  props: {
    stepNames: Array,
    currentStep: {
      type: Number,
      default: 1
    }
  },
  computed: {
    maxSteps () {
      return this.stepNames.length;
    },
    showNextButton () {
      return this.currentStep < this.maxSteps;
    },
    showPrevButton () {
      return this.currentStep > 1;
    }
  },
  methods: {
    prev () {
      this.$emit('goToStep', this.currentStep - 1);
    },
    next () {
      this.$emit('goToStep', this.currentStep + 1);
    },
    goToDashboard () {
      this.$router.push({ name: 'home' });
    }
  }
};
</script>

<style scoped lang="scss">
@import 'new-dashboard/styles/variables';

.footer {
  padding: 2em;
  background: $white;
}
</style>
