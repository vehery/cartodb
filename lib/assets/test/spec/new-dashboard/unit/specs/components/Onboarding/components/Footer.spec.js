import { shallowMount } from '@vue/test-utils';
import Footer from 'new-dashboard/components/Onboarding/components/Footer';

const $t = key => key;

describe('Footer.vue', () => {
  it('should render properly', () => {
    const footerComponent = createFooterComponent({
      propsData: {
        currentStep: 2
      }
    });

    expect(footerComponent).toMatchSnapshot();
  });

  it('should render go to dashboard button in last step', () => {
    const footerComponent = createFooterComponent({
      propsData: {
        currentStep: 3
      }
    });

    expect(footerComponent).toMatchSnapshot();
  });

  describe('Methods', () => {
    it('should emit goToStep event when clicking prev button', () => {
      const footerComponent = createFooterComponent({
        propsData: {
          currentStep: 2
        }
      });

      footerComponent.find('button.js-prev').trigger('click');

      expect(footerComponent.emitted().goToStep[0]).toEqual([1]);
    });

    it('should emit goToStep event when clicking next button', () => {
      const footerComponent = createFooterComponent({
        propsData: {
          currentStep: 1
        }
      });

      footerComponent.find('button.js-next').trigger('click');

      expect(footerComponent.emitted().goToStep[0]).toEqual([2]);
    });

    it('should call $router.push when going to home', () => {
      const $router = {
        push: jest.fn()
      };

      const footerComponent = createFooterComponent({
        propsData: {
          currentStep: 3
        },
        mocks: { $router }
      });

      footerComponent.find('button.js-goToDashboard').trigger('click');

      expect($router.push).toHaveBeenCalledWith({name: 'home'});
    });
  });
});

function createFooterComponent (overrides = {}) {
  const defaultSteps = [
    'Fake Step 1',
    'Fake Step 2',
    'Fake Step 3'
  ];

  const footerComponent = shallowMount(Footer, {
    ...overrides,
    propsData: {
      stepNames: defaultSteps,
      ...overrides.propsData
    },
    mocks: {
      $t,
      ...overrides.mocks
    }
  });

  return footerComponent;
}
