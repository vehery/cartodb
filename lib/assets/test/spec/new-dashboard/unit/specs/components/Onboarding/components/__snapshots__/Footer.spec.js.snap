// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.vue should render go to dashboard button in last step 1`] = `
<div class="footer">
  <div class="container u-flex u-flex__justify--end"><button class="button button--ghost js-prev">
      Wizards.footer.back
    </button>
    <!----> <button class="button is-primary js-goToDashboard">
      Wizards.footer.goToDashboard
    </button></div>
</div>
`;

exports[`Footer.vue should render properly 1`] = `
<div class="footer">
  <div class="container u-flex u-flex__justify--end"><button class="button button--ghost js-prev">
      Wizards.footer.back
    </button> <button class="button is-primary js-next">
      Wizards.footer.next
    </button>
    <!---->
  </div>
</div>
`;
